packages/types/src/ts-estree.ts

```diff
@@ -1496,13 +1493,13 @@ export interface TSMethodSignatureNonComputedName
 
 export interface TSModuleBlock extends BaseNode {
   type: AST_NODE_TYPES.TSModuleBlock;
-  body: ProgramStatement[];
+  body: Statement[];
 }
 
 export interface TSModuleDeclaration extends BaseNode {
   type: AST_NODE_TYPES.TSModuleDeclaration;
   id: Identifier | Literal;
-  body?: TSModuleBlock;
+  body?: TSModuleBlock | TSModuleDeclaration;
```

Comment:
we really need to fix this - https://github.com/typescript-eslint/typescript-eslint/issues/2573

Reply:
yes we should, but we should do that in separate PR

Reply:
if you look into no-unused-vars i had to add type guard against `TSModuleDeclaration`